MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -2156,6 +2109,36 @@ public static Person generateInstanceFromXML(Node wn, Campaign c, Version versio
                 retVal.setExpectedDueDate(retVal.getDueDate());
             }
 
+            //versions before 0.3.4 did not have proper clan phenotypes
+            if (version.isLowerThan("0.3.4") && c.getFaction().isClan()) {
```

Comment:
I wonder if anyone actually has saves from before 3.4. I would suggest moving this out into a separate method.

Reply:
Next part of my person changes following this and genealogy is going to be reworking the xml read/write into a simpler format. This will be part of doing so.